### ISSUE TITLE
Success callback is not triggered on non-bang transition then save

### ISSUE THREAD

**Issue by samphilipd** [aasm/aasm#268]
Here is an event I have defined on an ActiveRecord model

```
event :finalize do
  transitions from: :open, to: :finalized
  success do
    do_something
  end
end
```

If I call `model.finalize!` the success callback fires as expected.

However, I wanted to make sure to fail noisily if there were any validation failures, so I called it like this:

`model.finalize; model.save!`

In the second case, the success callback is never triggered even though the transition was successfully persisted.

As a workaround I am now making the transition like this:

`model.save!; model.finalize!`

which correctly triggers the callbacks and also fails noisily if the model isn't valid. But it triggers two database actions instead of one.

**Comment by alto** [comment]
@samphilipd I agree. As you can see in #262 the usage of `save` instead of `save!` already bit other developers as well.

What do you think of the suggested solution (use `save` if `whiny_transitions: false, and `save!` if `whiny_transitions: true`)?

**Comment by samphilipd** [comment]
@alto thanks for looking into this, I think we need a separate option though.

IMO for bang methods, `save!` should be the default. How about a default option like `raise_on_validation_failure: true`. I can write a patch for this if you like.

**Comment by dannytip** [comment]
This has just bitten me too. I'm simply wanting to do: `model.finalize; model.save`

But no callbacks are firing.  Ive tried `success`, `after_commit` and various other.

**Comment by fladson** [comment]
Is this solved? I'm trying to ensure a method call on `success` but it isn't being called, I also have tried `after_commit`, `after_transaction` with no success, it just works with `after`

**Comment by art-solopov** [comment]
Doesn't work for me on 4.11.1.

**Comment by pedros007** [comment]
@samphilipd you can make ActiveRecordCallbacks "fail noisily" with AASM bang transitions when you set the `whiny_persistence: true` option, like so:

```ruby
aasm column: :workflow_state, whiny_persistence: true do
  ...
end
```

Now if `finalize` should raise, the AASM transition `my_object.finalize!` will raise, like so:

```ruby
ActiveRecord::RecordInvalid: Validation failed: whatever_the_validation_message_is
```
